src/index.ts

```
@@ -0,0 +1,116 @@
+import * as core from '@actions/core'
+import * as yaml from 'yaml'
+import * as fs from "fs";
+import fetch from 'node-fetch';
+import path from 'path';
+import {context} from "@actions/github";
+import * as github from '@actions/github';
+
+const run = async () => {
+    const baseUrl = `https://api.roadie.so`;
+    const catalogInfoPath = core.getInput('path')
+    const apiKey = core.getInput('roadie-api-key')
+
+    const githubToken = core.getInput('github-token');
+    const octokit = github.getOctokit(githubToken)
+
+    const res = await octokit.request( 'GET /repos/{owner}/{repo}/compare/{basehead}', {
+        owner: context.payload.organization?.login,
+        repo: context.payload.repository?.name,
+        basehead: `${context.payload.pull_request?.base.ref}...${context.payload.pull_request?.head.ref}`,
+        per_page: 100,
+    } );
+    const filesChanged = res.data.files.map((f: any) => f?.filename)
+    console.log(`${filesChanged.length} changed files`)
+
+    if(apiKey === '') {
```

Maybe check this before making octokit calls so we can fail fast.